src/components/RandomLoanSelector/RandomLoanSelector.vue

```diff
@@ -22,18 +21,18 @@
 					ref="randomLoansCarousel"
 					:multiple-slides-visible="true"
 					slides-to-scroll="visible"
-					:slide-max-width="carouselCardWidth"
+					:slide-max-width="singleSlideWidth"
 					@interact-carousel="onInteractCarousel"
 				>
 					<template v-for="(loan, index) in randomLoans" #[`slide${index}`]>
-						<minimal-loan-card
-							:key="index"
-							:class="`minimal-loancard-${index}`"
-							:loan="loan"
-							category-set-id="random-loans"
-							:card-number="index"
-							@refreshtotals="$emit('refreshtotals')"
-							@updating-totals="$emit('updating-totals', $event)"
```

Comment:
Can we please restore this functionality by emitting an `updating-totals` event from the `addToBasket` method within KvClassicLoanCardContainer? This emit would be at the beginning of that method along with `this.isAdding = true;`, before the mutation occurs.

As it stands there is no loading state triggered until after the add to basket mutation has completed and emitted the addToBasket event. Ideally, we trigger a loading state for the page on click. The `refreshTotals` method you've set below will handle removing the loading state.

Reply:
ok, that makes sense. thanks for checking!